tox.ini

```diff
@@ -5,12 +5,8 @@ isolated_build = True
 minversion = 3.3.0
 
 [testenv]
-deps =
-    dev: joblib @ git+https://github.com/joblib/joblib.git
```

Comment:
This line needs to be kept so that it can be used by the py-dev job in `.github/workflows/test.yml`.

Reply:
oh! So 

- this line means that it defines `py-dev` tox environment with that extra dependency
- as long as our CI passes, we can upgrade joblib to not be `~= 1.1` but rather `>= 1.1.0, <1.4` ATM or even without upper limit and just react on breakage with dev joblib and if needed preemptively having it fixed etc?

Reply:
meanwhile -- reverted that change, force pushed cleaned up 2nd commit

Reply:
> we can upgrade joblib to not be `~= 1.1` but rather `>= 1.1.0, <1.4`

What would be the point of that?  Note that `~= 1.1` means `>= 1.1, <2.0`, not `>= 1.1, <1.2`.

Reply:
you are right, I misunderstood it as `<1.2`.
meanwhile -- we turned CI green ;-)